code/modules/fishing/fishing_rod.dm

```diff
@@ -118,11 +118,14 @@
 	return hook?.reason_we_cant_fish(target_fish_source)
 
 /obj/item/fishing_rod/proc/consume_bait(atom/movable/reward)
-	if(!reward)
+	// catching things that aren't fish or alive mobs doesn't consume baits.
+	if(!reward || !bait)
```

Comment:
```suggestion
	if(isnull(reward) || isnull(bait))
```
I believe this is marginally faster (unimportant) but more importantly it's clearer